### PR TITLE
Adjust ioctl function of cellular

### PR DESCRIPTION
**Pull request by pstys**
## Summary
Add SIOCGCELLNETDEV
Give a uint8_t[136] for cellular to save data
## Impact
No
## Testing
Selftest

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: net/netdev/netdev_ioctl.c [hunk 1]
```diff
@@ -544,6 +548,51 @@ static int netdev_pktradio_ioctl(FAR struct socket *psock, int cmd,
...
+                             FAR struct icellreq *req)
+{
+  FAR struct net_driver_s *dev = NULL;
+  int ret = -ENOTTY;
+
+  net_lock();
+
+  if (_CELLIOCVALID(cmd))
+    {
+      ninfo("cmd: %d\n", cmd);
```
Comment (anchao) [inline net/netdev/netdev_ioctl.c:579]: move syslog out of net_lock()

Comment (pstys) [reply net/netdev/netdev_ioctl.c]: Done

File: net/netdev/netdev_ioctl.c [hunk 2]
```diff
@@ -544,6 +548,51 @@ static int netdev_pktradio_ioctl(FAR struct socket *psock, int cmd,
...
+      dev = netdev_findbyname(req->ifr_name);
+      if (dev && dev->d_ioctl)
+        {
+          ret = dev->d_ioctl(dev, cmd, (unsigned long)(uintptr_t)req);
+        }
+      else
+        {
+          ret = -ENOTTY;
```
[the remark this answers is not in the document]
Comment (pstys) [reply net/netdev/netdev_ioctl.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
Let's ignore the ci broken by https://github.com/apache/nuttx/pull/7832

**Pull request merged**